homeassistant/components/integration/sensor.py

```diff
@@ -273,6 +273,7 @@ async def async_added_to_hass(self) -> None:
                 self._state,
                 self._last_valid_state,
             )
+            self.async_write_ha_state()
```

Comment:
What's the problem were trying to solve? The entity will always write its state at the end of entity addition by default.

Reply:
From my understanding of the linked issue the state is not updated until a new state change (that calls sync_ha_state)

Reply:
This change won't solve the problem. Here's the code:

https://github.com/home-assistant/core/blob/11c5e3534aeae67a3e0c3650030a00175f3f1672/homeassistant/helpers/entity.py#L986-L990